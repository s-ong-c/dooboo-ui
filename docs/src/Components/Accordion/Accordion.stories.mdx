import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';

<Meta title="Components|Accordion" />

# Accordion

> Simple drop down item component for react-native. This component supports drop down toggle with animation.<br/>

![image](https://user-images.githubusercontent.com/27461460/47951961-a7a7e500-dfab-11e8-9189-86c0eddb6e12.gif)

## Props

|                    | necessary | types              | default               |
| ------------------ | --------- | ------------------ | --------------------- |
| data               | âœ“         | array              | `Default Data`        |
| isAnimated         |           | boolean            | `true`                |
| collapseOnStart    |           | boolean            | `false`               |
| animDuration       |           | number             | `300`                 |
| activeOpacity      |           | number             | `1`                   |
| toggleElement      |           | React.ReactElement | `default arrow image` |
| accordionItemStyle |           | ViewStyle          | ``                    |
| titleStyle         |           | ViewStyle          | ``                    |
| titleStyle         |           | ViewStyle          | ``                    |

## Installation

```sh
yarn add @dooboo-ui/core
```

## Getting started

- Import

```javascript
import { Accordion } from '@dooboo-ui/core';
```

## Data

```javascript
state = {
const data = [
  {
    title: {
      name: <StyledTitle>Defualt-title-01</StyledTitle>,
    },
    bodies: [
      {
        name: <StyledItem>Default body01</StyledItem>,
      },
      {
        name: <StyledItem>Default body02</StyledItem>,
      },
    ],
  },
  {
    title: {
      name: <StyledTitle>Defualt-title-02</StyledTitle>,
    },
    bodies: [
      {
        name: <StyledItem>Default body01</StyledItem>,
      },
      {
        name: <StyledItem>Default body02</StyledItem>,
      },
    ],
  },
];
}
```

## Usage

```tsx
<Accordion
  data={accordionData}
  isAnimate={true}
  collapseOnStart={false}
  animDuration={300}
  activeOpacity={1}
/>
```
