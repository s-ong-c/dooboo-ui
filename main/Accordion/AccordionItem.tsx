import {
  Animated,
  LayoutChangeEvent,
  View,
  ViewStyle,
} from 'react-native';
import React, { FC, useCallback, useEffect, useRef, useState } from 'react';

import styled from 'styled-components/native';

const Container = styled.View`
  background-color: transparent;
  overflow: hidden;
  width: 300px;
`;

const TitleContainer = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  background-color: #141414;
  height: 50px;
  border-bottom-width: 1px;
  border-bottom-color: white;
  z-index: 10;
`;

const ToggleIndicator = styled.View`
  position: absolute;
  right: 20px;
`;

const StyledItem = styled.View`
  flex-direction: row;
  align-items: center;
  width: 100%;
  padding: 20px 40px;
  border-bottom-width: 1px;
  border-bottom-color: lightgray;
`;

enum DirectionIndicator {
  'UP' = 'up',
  'DOWN' = 'down',
}

type titleType = {
  leftElement?: React.ReactElement;
  name: React.ReactElement;
  rightElement?: React.ReactElement;
};

type bodyType = {
  leftElement?: React.ReactElement;
  name: React.ReactElement;
  rightElement?: React.ReactElement;
};

type datumType = {
  title: titleType;
  bodies: Array<bodyType>;
};
interface Props {
  testID: string;
  datum: datumType;
  shouldAnimate?: boolean;
  collapseOnStart: boolean;
  animDuration?: number;
  activeOpacity?: number;
  toggleElement?: React.ReactElement;
  accordionItemStyle?: ViewStyle;
  titleStyle?: ViewStyle;
  bodyStyle?: ViewStyle;
}

let layoutHeight = 0;

const AccordionItem: FC<Props> = (props) => {
  const {
    testID,
    datum,
    shouldAnimate,
    collapseOnStart,
    animDuration,
    activeOpacity,
    toggleElement,
    accordionItemStyle,
    titleStyle,
    bodyStyle,
  } = props;

  const animValue = useRef(new Animated.Value(0)).current;

  const [opened, setItemVisible] = useState<boolean>(collapseOnStart);
  const [bodyMounted, setBodyMounted] = useState<boolean>(false);

  const [bodyHeight, setBodyHeight] = useState<number>(1);

  const [directionIndicator, setDirectionIndicator] = useState(DirectionIndicator.DOWN);

  const handleBodyLayout = (e: LayoutChangeEvent): void => {
    if (bodyMounted) return;
    const { height } = e.nativeEvent.layout;
    layoutHeight = height;
    setBodyMounted(true);
    setBodyHeight(height);
  };

  const handleVisibleState = (): void => {
    setItemVisible(!opened);
  };

  useEffect((): void => {
    if (bodyMounted) {
      animValue.setValue(opened ? layoutHeight : 0);
    }
  }, [bodyMounted]);

  useEffect((): void => {
    if (bodyHeight === layoutHeight) {
      Animated.timing(animValue, {
        toValue: 0,
        duration: animDuration || 300,
        useNativeDriver: true,
        delay: 10,
      }).start();

      setDirectionIndicator(DirectionIndicator.UP);
    }
  }, [bodyHeight]);

  useEffect((): void => {
    if (shouldAnimate) {
      if (!opened) {
        setBodyHeight(layoutHeight);
        return;
      }

      // setBodyHeight(0);
      Animated.timing(animValue, {
        toValue: 1,
        duration: animDuration || 300,
        useNativeDriver: true,
      }).start(() => {
        setBodyHeight(0);
      });

      setDirectionIndicator(DirectionIndicator.DOWN);
    } else {
      const targetValue = opened ? 1 : 0;
      animValue.setValue(targetValue);

      if (opened) {
        setDirectionIndicator(DirectionIndicator.UP);
        return;
      }

      setDirectionIndicator(DirectionIndicator.DOWN);
    }
  }, [opened]);

  return (
    <Container
      style={accordionItemStyle}
    >
      <TitleContainer
        testID={`title_${testID}`}
        onPress={handleVisibleState}
        activeOpacity={activeOpacity}
        style={titleStyle}
      >
        {datum.title.leftElement && datum.title.leftElement}
        {datum.title.name}
        {datum.title.rightElement ? (
          datum.title.rightElement
        ) : (
          <ToggleIndicator
            style={{
              transform: [
                directionIndicator === 'up'
                  ? { rotate: '180deg' }
                  : { rotate: '0deg' },
              ],
            }}>
            {toggleElement || null}
          </ToggleIndicator>
        )}
      </TitleContainer>
      <Animated.View
        testID={`body_${testID}`}
        style={{
          height: !bodyMounted
            ? undefined
            : bodyHeight,
          transform: [
            {
              translateY: animValue.interpolate({
                inputRange: [0, 1],
                outputRange: [0, -bodyHeight],
              }),
            },
          ],
        }}
        onLayout={handleBodyLayout}
      >
        {datum.bodies.map((body, bodyKey) => {
          return (
            <StyledItem key={bodyKey} style={bodyStyle}>
              {body.leftElement && body.leftElement}
              {body.name}
              {body.rightElement && body.rightElement}
            </StyledItem>
          );
        })}
      </Animated.View>
    </Container>
  );
};

export default AccordionItem;
