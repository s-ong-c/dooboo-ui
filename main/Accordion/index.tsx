import React, { FC } from 'react';
import AccrordionItem from './AccordionItem';
import { ViewStyle } from 'react-native';
import styled from 'styled-components/native';

const Container = styled.View`
  flex-direction: column;
  align-items: center;
`;

type titleType = {
  leftElement?: React.ReactElement;
  name: React.ReactElement;
  rightElement?: React.ReactElement;
};

type bodyType = {
  leftElement?: React.ReactElement;
  name: React.ReactElement;
  rightElement?: React.ReactElement;
};

type datumType = {
  title: titleType;
  bodies: Array<bodyType>;
};

interface Props {
  data: Array<datumType>;
  isAnimated?: boolean;
  collapseOnStart: boolean;
  animDuration?: number;
  activeOpacity?: number;
  toggleElement?: React.ReactElement;
  accordionItemStyle?: ViewStyle;
  titleStyle?: ViewStyle;
  bodyStyle?: ViewStyle;
}

const Accordion: FC<Props> = (props) => {
  const {
    data,
    isAnimated: shouldAnimate,
    collapseOnStart,
    animDuration,
    activeOpacity,
    toggleElement,
    accordionItemStyle,
    titleStyle,
    bodyStyle,
  } = props;

  return (
    <Container>
      {data.map((datum, titleKey) => {
        return (
          <AccrordionItem
            testID={`${titleKey}`}
            key={titleKey}
            datum={datum}
            shouldAnimate={shouldAnimate}
            collapseOnStart={collapseOnStart}
            animDuration={animDuration}
            activeOpacity={activeOpacity}
            toggleElement={toggleElement}
            accordionItemStyle={accordionItemStyle}
            titleStyle={titleStyle}
            bodyStyle={bodyStyle}
          />
        );
      })}
    </Container>
  );
};
export default Accordion;
