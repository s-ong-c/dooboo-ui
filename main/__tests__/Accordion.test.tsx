import React, { ReactElement } from 'react';
import {
  RenderResult,
  act,
  fireEvent,
  render,
} from '@testing-library/react-native';
import Accordion from '../Accordion';
import { Text } from 'react-native';

let props: any;
let component: ReactElement;
let testingLib: RenderResult;
const data = [
  {
    title: {
      name: <Text>Defualt-title-01</Text>,
    },
    bodies: [
      {
        name: <Text>Default body01</Text>,
      },
      {
        name: <Text>Default body02</Text>,
      },
    ],
  },
  {
    title: {
      name: <Text>Defualt-title-02</Text>,
    },
    bodies: [
      {
        name: <Text>Default body01</Text>,
      },
      {
        name: <Text>Default body02</Text>,
      },
    ],
  },
];

const createTestProps = (obj?: Record<string, unknown>): Record<string, unknown> => ({
  ...obj,
});

describe('[Accordion] render test', () => {
  it('should render without crasing', () => {
    props = createTestProps({ data: data });
    component = <Accordion {...props}/>;
    testingLib = render(component);

    expect(testingLib.baseElement).toMatchSnapshot();
  });

  it('should render collapsed when collapseOnStart props is true', () => {
    props = createTestProps({
      collapseOnStart: true,
      data: data,
    });
    component = <Accordion {...props}/>;
    testingLib = render(component);

    expect(testingLib.baseElement).toMatchSnapshot();
  });

  it('should operate animation when isAnimated props is true', () => {
    jest.useFakeTimers();
    props = createTestProps({
      isAnimated: true,
      data: data,
    });
    component = <Accordion {...props}/>;
    testingLib = render(component);
    jest.runAllTimers();

    expect(testingLib.baseElement).toMatchSnapshot();
  });

  it('should adjust duration of animation depends on animDuration props value', () => {
    props = createTestProps({
      animDuration: 500,
      data: data,
    });
    component = <Accordion {...props}/>;
    testingLib = render(component);

    expect(testingLib.baseElement).toMatchSnapshot();
  });
});

describe('[Accordion] event test', () => {
  beforeEach(() => {
    props = createTestProps({
      data: data,
    });
    component = <Accordion {...props}/>;
    testingLib = render(component);
  });

  it('should trigger onLayout event when itemBody rendered', () => {
    const { getByTestId } = testingLib;
    const itemTitle = getByTestId('body_0');

    act(() => {
      fireEvent.layout(itemTitle, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });
  });

  it('should trigger press event when clicking title', () => {
    act(() => {
      fireEvent.press(testingLib.getByTestId('title_0'));
    });
  });
});
