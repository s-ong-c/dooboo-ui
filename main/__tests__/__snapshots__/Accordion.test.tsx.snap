// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Accordion] render test should adjust duration of animation depends on animDuration props value 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "zIndex": 10,
              },
            ]
          }
          testID="title_0"
        >
          <Text>
            Defualt-title-01
          </Text>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                },
              ]
            }
          />
        </>
        <View
          style={
            Object {
              "height": undefined,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          testID="body_0"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body01
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body02
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "zIndex": 10,
              },
            ]
          }
          testID="title_1"
        >
          <Text>
            Defualt-title-02
          </Text>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                },
              ]
            }
          />
        </>
        <View
          style={
            Object {
              "height": undefined,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          testID="body_1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body01
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body02
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Accordion] render test should operate animation when isAnimated props is true 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "zIndex": 10,
              },
            ]
          }
          testID="title_0"
        >
          <Text>
            Defualt-title-01
          </Text>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "180deg",
                    },
                  ],
                },
              ]
            }
          />
        </>
        <View
          style={
            Object {
              "height": undefined,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          testID="body_0"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body01
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body02
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "zIndex": 10,
              },
            ]
          }
          testID="title_1"
        >
          <Text>
            Defualt-title-02
          </Text>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "180deg",
                    },
                  ],
                },
              ]
            }
          />
        </>
        <View
          style={
            Object {
              "height": undefined,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          testID="body_1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body01
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body02
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Accordion] render test should render collapsed when collapseOnStart props is true 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "zIndex": 10,
              },
            ]
          }
          testID="title_0"
        >
          <Text>
            Defualt-title-01
          </Text>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "180deg",
                    },
                  ],
                },
              ]
            }
          />
        </>
        <View
          style={
            Object {
              "height": undefined,
              "transform": Array [
                Object {
                  "translateY": -1,
                },
              ],
            }
          }
          testID="body_0"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body01
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body02
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "zIndex": 10,
              },
            ]
          }
          testID="title_1"
        >
          <Text>
            Defualt-title-02
          </Text>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "180deg",
                    },
                  ],
                },
              ]
            }
          />
        </>
        <View
          style={
            Object {
              "height": undefined,
              "transform": Array [
                Object {
                  "translateY": -1,
                },
              ],
            }
          }
          testID="body_1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body01
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body02
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Accordion] render test should render without crasing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "zIndex": 10,
              },
            ]
          }
          testID="title_0"
        >
          <Text>
            Defualt-title-01
          </Text>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                },
              ]
            }
          />
        </>
        <View
          style={
            Object {
              "height": undefined,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          testID="body_0"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body01
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body02
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "zIndex": 10,
              },
            ]
          }
          testID="title_1"
        >
          <Text>
            Defualt-title-02
          </Text>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                },
              ]
            }
          />
        </>
        <View
          style={
            Object {
              "height": undefined,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          testID="body_1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body01
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "lightgray",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingBottom": 20,
                  "paddingLeft": 40,
                  "paddingRight": 40,
                  "paddingTop": 20,
                  "width": "100%",
                },
              ]
            }
          >
            <Text>
              Default body02
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
